cylc/flow/unicode_rules.py

```diff
@@ -103,6 +106,29 @@ def not_starts_with(*chars):
     )
 
 
+def not_contains_colon_unless_starts_with(*chars):
+    """Restrict use of colons.
+
+    Example:
+        >>> regex, message = not_contain_colon_unless_starts_with(
+                'INFO', 'WARNING')
+        >>> message
+        'cannot contain a colon unless starts with: ``INFO``, ``WARNING``'
+        >>> bool(regex.match('Foo: bar'))
+        False
+        >>> bool(regex.match('INFO: Foo: bar'))
+        True
```

Comment:
Do these automatically generate unit tests?

Reply:
Pytest is configured to run any doctests found in the source code.